src/assets/scripts/client/aircraft/AircraftCommander.js

```diff
@@ -308,19 +308,27 @@ export default class AircraftCommander {
             return [false, `unable to hold at unknown fix ${fixName}`];
         }
 
-        let inboundHeading = fixModel.positionModel.bearingFromPosition(aircraft.positionModel);
+        const holdParameters = {};
 
-        // Radial is given as the outbound course, so it needs to be inverted
+        // Instead of using DEFAULT_HOLD_PARAMETERS here, we only pass the values
+        // provided by the player. These then are used to patch the _holdParameters as
+        // specified in the `WaypointModel`
+        if (turnDirection !== null) {
+            holdParameters.turnDirection = turnDirection;
+        }
+        if (legLength !== null) {
+            holdParameters.legLength = legLength;
+        }
         if (radial !== null) {
-            inboundHeading = radians_normalize(degreesToRadians(radial) + Math.PI);
+            // Radial is given as the outbound course, so it needs to be inverted
+            holdParameters.inboundHeading = radians_normalize(degreesToRadians(radial) + Math.PI);
+        } else {
+            // As radial has not been explicitly requested, we need to pass a "fallback"
+            // inboundHeading, as it's possible that a default inboundHeading doesn't exist for
+            // the `WaypointModel`s _holdParameters (eg. if the procedure JSON has no holds)
+            holdParameters.fallbackHeading = fixModel.positionModel.bearingFromPosition(aircraft.positionModel);
```

Comment:
Sure, the reason I implemented it like this was because I wanted the behaviour to replicate the existing behaviour (ie not breaking things).

`EZY97GP rr bpk..gwe16 hold gwe16`
![image](https://user-images.githubusercontent.com/3430117/68075702-2728e980-fda3-11e9-802d-b62f77202ce4.png)


I'll have a look at trying to get it to work as you describe, making sure that the readback will be also be correct if the hold fix requested is later on in the procedure.

Reply:
So, this behaviour is corrected:
1. `AircraftCommander.runHold` calculates the `fallbackHeading` to use as the heading from the aircraft's current position to the holding fix (this is the existing behaviour)
2. `RouteModel.activateHoldForWaypointName` then calculates the inbound course to the holding fix from the preceding fix and uses this as `fallbackHeading`
If there is no preceding waypoint, then it uses the `fallbackHeading` from (1)
3. This is then passed all the way down to `WaypointModel.setHoldParameters`, where it is used if no default `inboundHeading` value is present.